Code/Source/RobotImporter/URDF/CollidersMaker.cpp

```diff
@@ -0,0 +1,346 @@
+                entity->CreateComponent(LmbrCentral::EditorBoxShapeComponentTypeId);
+                auto boxGeometry = std::dynamic_pointer_cast<urdf::Box>(geometry);
+                entity->Activate();
+                LmbrCentral::BoxShapeComponentRequestsBus::Event(
+                    entityId,
+                    &LmbrCentral::BoxShapeComponentRequests::SetBoxDimensions,
+                    URDF::TypeConversions::ConvertVector3(boxGeometry->dim));
+                entity->Deactivate();
+            }
+            break;
+        case urdf::Geometry::CYLINDER:
+            {
+                entity->CreateComponent(LmbrCentral::EditorCylinderShapeComponentTypeId);
+                auto cylinderGeometry = std::dynamic_pointer_cast<urdf::Cylinder>(geometry);
+                entity->Activate();
+                LmbrCentral::CylinderShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::CylinderShapeComponentRequests::SetHeight, cylinderGeometry->length);
+                LmbrCentral::CylinderShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::CylinderShapeComponentRequests::SetRadius, cylinderGeometry->radius);
+                entity->Deactivate();
+            }
+            break;
+        default:
+            AZ_Warning("AddCollider", false, "Unsupported collider geometry type, %d", geometry->type);
+            break;
+        }
+    }
+    AZ::IO::Path CollidersMaker::GetFullURDFMeshPath(AZ::IO::Path modelPath, AZ::IO::Path meshPath)
+    {
+        modelPath.RemoveFilename();
+        AZ::StringFunc::Replace(meshPath.Native(), "package://", "", true, true);
+        modelPath /= meshPath;
+
+        return modelPath;
+    }
+
+    void CollidersMaker::ProcessMeshes(BuildReadyCallback notifyBuildReadyCb)
+    {
+        m_buildThread = AZStd::thread(
+            [this, notifyBuildReadyCb]()
+            {
+                AZ_Printf("CollisionMaker", "Waiting for URDF assets...");
+                while (!m_meshesToBuild.empty() && !m_stopBuildFlag)
+                {
+                    {
+                        AZStd::lock_guard lock{ m_buildMutex };
+                        for (auto iter = m_meshesToBuild.begin(); iter != m_meshesToBuild.end(); iter++)
+                        {
+                            AZ::Data::AssetId assetId;
+                            AZ::Data::AssetType assetType = AZ::AzTypeInfo<PhysX::Pipeline::MeshAsset>::Uuid();
+                            AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                                assetId, &AZ::Data::AssetCatalogRequests::GetAssetIdByPath, iter->Native().c_str(), assetType, false);
+                            if (assetId.IsValid())
+                            {
+                                AZ_Printf("CollisionMaker", "Asset %s found and valid...", iter->Native().c_str());
+                                m_meshesToBuild.erase(iter--);
+                            }
+                        }
```

Comment:
You can use the [AZStd::erase_if](https://en.cppreference.com/w/cpp/container/vector/erase2) function to remove the paths from the `m_meshesToBuild` container.

nitpick: The AZ::IO::Path class has a `.c_str()` function on it which invokes c_str on the underlying string type.
So `.Native().c_str()` can be replace with just `c_str()`

```suggestion
                        auto eraseFoundMesh = [](const AZ::IO::Path& meshPath)
                        {
                            AZ::Data::AssetId assetId;
                            AZ::Data::AssetType assetType = AZ::AzTypeInfo<PhysX::Pipeline::MeshAsset>::Uuid();
                            AZ::Data::AssetCatalogRequestBus::BroadcastResult(
                                assetId, &AZ::Data::AssetCatalogRequests::GetAssetIdByPath, meshPath.c_str(), assetType, false);
                            if (assetId.IsValid())
                            {
                                AZ_Printf("CollisionMaker", "Asset %s found and valid...", meshPath.c_str());
                                return true; // return true to erase the mesh path
                            }
                            
                            return false;
                        };

                        AZStd::erase_if(m_meshesToBuild, AZStd::move(eraseFoundMesh ));
```